internal/supervisor/installation.go

```diff
@@ -394,7 +399,7 @@ func (s *InstallationSupervisor) waitForClusterInstallationStable(installation *
 
 	if len(clusterInstallations) == stable {
 		logger.Info("Finished creating cluster installation")
-		return s.configureInstallationDNS(installation, logger)
+		return model.InstallationStateCreationFinalTasks
```

Comment:
Are we unable to proceed with the next step directly here? Before, we would always try to call the function associated with the next state directly after one state finished. This isn't a hard-and-fast rule that we always follow, but I am just wondering if we can continue to do that here and a few other places in the new code?

If we have a valid reason to not do so that is obviously fine as well, but I don't see an immediate reason why we can't continue.

Reply:
Hey @gabrieljackson the reason I changed it to follow this pattern is to have a clear understanding of which state you are in. For example if you call directly the final tasks, in the first iteration of the function although it will be running the code of state InstallationStateCreationFinalTasks the official state will still be the previous one. Only if the first iteration fails for some reason it will move the state to InstallationStateCreationFinalTasks and try again. Don't we want the state to change before it moves to the next step?

Reply:
That's a good question and you are right that ideally we know what state the object is in as it is worked on. So the trade off right now of not moving through the states directly by calling the next function is that the supervisor will go to sleep after every state code flow. In other words, where as we might have been able to do 3 out of 5 state processes in a row and then need some sleep time on the supervisor, this will sleep after each step. This, obviously, is not really a problem. It just means that the time to get the installation to stable will be longer than it has to.

All that said, I am fine with this. I just wanted to ask if we were returning after each step for a reason or not. I also think it is possible to have the best of both worlds in the future. We should be able to update state as we go while also not stopping until we have to, but that is way beyond the scope of this ticket.

Reply:
As said, its good to have the best of both worlds but for now making the installation creation faster is more important. I moved to call directly the functions rather than setting the state first.